chrome/content/zotero/xpcom/zotero.js

```diff
@@ -2063,3 +2063,79 @@ Zotero.JSON = new function() {
 		return JSON.parse(arg);
 	}
 }
+
+/**
+ * Manages preference panes.
+ */
+Zotero.PreferencePanes = {
+	builtInPanes: Object.freeze([
+		{
+			id: 'zotero-prefpane-general',
+			label: 'zotero.preferences.prefpane.general',
+			image: 'chrome://zotero/skin/prefs-general.png',
+			src: 'chrome://zotero/content/preferences/preferences_general.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_general.js']
+		},
+		{
+			id: 'zotero-prefpane-sync',
+			label: 'zotero.preferences.prefpane.sync',
+			image: 'chrome://zotero/skin/prefs-sync.png',
+			src: 'chrome://zotero/content/preferences/preferences_sync.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_sync.js']
+		},
+		{
+			id: 'zotero-prefpane-export',
+			label: 'zotero.preferences.prefpane.export',
+			image: 'chrome://zotero/skin/prefs-export.png',
+			src: 'chrome://zotero/content/preferences/preferences_export.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_export.js']
+		},
+		{
+			id: 'zotero-prefpane-cite',
+			label: 'zotero.preferences.prefpane.cite',
+			image: 'chrome://zotero/skin/prefs-styles.png',
+			src: 'chrome://zotero/content/preferences/preferences_cite.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_cite.js']
+		},
+		{
+			id: 'zotero-prefpane-advanced',
+			label: 'zotero.preferences.prefpane.advanced',
+			image: 'chrome://zotero/skin/prefs-advanced.png',
+			src: 'chrome://zotero/content/preferences/preferences_advanced.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_advanced.js']
+		},
+		{
+			id: 'zotero-subpane-reset-sync',
+			parent: 'zotero-prefpane-sync',
+			label: 'zotero.preferences.subpane.resetSync',
+			src: 'chrome://zotero/content/preferences/preferences_sync_reset.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_sync.js']
+		}
+	]),
+
+	pluginPanes: [],
+
+	/**
+	 * Register a pane to be displayed in the preferences. The pane XHTML (`src`)
+	 * is loaded as a fragment, not a full document, with XUL as the default
+	 * namespace and (X)HTML tags available under `html:`.
+	 *
+	 * @param {Object} options
+	 * @param {String} options.id Must be unique
+	 * @param {String} [options.parent] ID of parent pane (if provided, pane is hidden from the sidebar)
+	 * @param {String} [options.label] A DTD/.properties key (optional for panes with parents)
+	 * @param {String} [options.rawLabel] A raw string to use as the label if options.label is not provided
+	 * @param {String} [options.image] URI of an icon (displayed in the navigation sidebar)
```

Comment:
Are we not doing anything for 2x images? Even for our panes? I suspect we never had 2x icons for those, and so this has no effect for our icons at the moment, but we'll definitely want to support multiple icon sizes for plugins.

For plugins we could theoretically skip both this by just having plugins pass their id, and we could get the images from the [`icons`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/icons) property in manifest.json by adding a function to plugins.js. We could also automatically use the name as the label but let plugins override with `label` in case the extension name is too long (e.g., "Zutilo" instead of "Zutilo Utility for Zotero", though honestly that's probably a holdover from Zotero for Firefox days and should just be changed).

Reply:
>Are we not doing anything for 2x images? Even for our panes?

We're not, and we never did have 2x icons for those. Latest commit should do the trick (but I don't have a plugin to test with yet).

Reply:
I agree that using the plugin name as label should be fine in almost all cases, but I want to leave things more open for a hypothetical complex plugin - a Jurism rework? - that might need to register multiple panes, that sort of thing.